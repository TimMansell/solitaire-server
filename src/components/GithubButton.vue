<template>
  <Button :is-stacked="isStacked" @click="gotoUrl" data-test="github-btn">
    <FontAwesomeIcon :icon="githubIcon" />
  </Button>
</template>

<script>
import { faGithub } from '@fortawesome/free-brands-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
import Button from './Button.vue';

export default {
  name: 'GithubButton',
  components: {
    Button,
    FontAwesomeIcon,
  },
  props: {
    isStacked: {
      type: Boolean,
      default: false,
    },
  },
  data() {
    return {
      githubIcon: faGithub,
    };
  },
  methods: {
    gotoUrl() {
      window.open('https://github.com/TimMansell/vue-solitaire', '_blank');
    },
  },
};
</script>
