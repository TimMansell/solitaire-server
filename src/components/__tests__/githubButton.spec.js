import { shallowMount } from '@vue/test-utils';
import GithubButton from '@/components/GithubButton.vue';

describe('GithubButton.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(GithubButton);

    expect(wrapper).toMatchSnapshot();
  });
});
