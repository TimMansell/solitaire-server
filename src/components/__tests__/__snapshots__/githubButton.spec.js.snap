// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubButton.vue matches snapshot 1`] = `
<button-stub type="default" size="md" data-test="github-btn" title="Visit Github Repository">
  <fontawesomeicon-stub icon="[object Object]"></fontawesomeicon-stub>
</button-stub>
`;
